### ISSUE TITLE
False positive "No reference to the subscription is kept"

### ISSUE THREAD

**Issue by KirillGorbach** [littlerobots/rxlint#11]
When I assign a variable in kotlin using if condition, the leaking subscription is falsely determined:
```kotlin
val subscription = if (condition) {
getMyRequestSingle().subscribe()
} else {
...
}
```
```.subscribe()``` provides warning "No reference to the subscription is kept", but the subscription is not really leaking
Using version 1.7.8

**Comment by hvisser** [comment]
I assume you are also returning a subscription in your else block right? It could be that this needs some specific handling. I haven't been using rxlint for a couple of years now so a PR to fix this would be welcome as I probably don't have the time for a fix at the moment.
